sympy/calculus/singularities.py

```diff
@@ -109,7 +110,47 @@ def singularities(expression, symbol, domain=None):
 ###########################################################################
 
 
+@deprecated(
+    "monotonicity_helper is considered a private function, use the direct"
+    " function for checking instead", deprecated_since_version="1.12",
+    active_deprecations_target="monotonicity-helper"
+)
 def monotonicity_helper(expression, predicate, interval=S.Reals, symbol=None):
+    """
+    .. deprecated:: 1.12
```

Comment:
This should probably be 1.13 unless you plan to backport this PR to the 1.12 branch that I just created.

Reply:
Ahh, I checked and was a bit confused about the release status.

Reply:
I haven't quite finished setting everything up on the branches: #24776

Reply:
Maybe we should try to have some automatic backporting? So if a PR is marked 1.12 it will be backported upon merging? I think https://github.com/meeseeksbox is a rather good option. 

Since I just pushed a number of doc-fixes it can make even more sense for those, rather than having to (more or less) manually backport stuff. Also convenient for a bugfix releases.

Reply:
It would be good to make it easier but it's quite unusual that anything really needs to be backported.

It's possible that making it easier would mostly just mean that things get backported unnecessarily.

Reply:
Could be. But depends on how one defines unnecessarily.

I guess it is primarily used for things one would like to get in either after the fork (bug fixes from RCs, documentation improvements) or if one tend to release regression fixes  (which we rarely do).

Right now there are not that many PRs marked 1.12 though.

Reply:
#24800 is a typical thing that could be backported at this stage if it was reasonably easy. Minor thing, but makes sense to get in. (And I realize that "it will never break anything"-stuff is not always good to get in with short notice...)

Reply:
SymPy's releases are basically just a snapshot of the master branch at some more or less arbitrarily chosen point in time. If there is any value in having a release branch then it is so that we can limit what changes are applied to it. In that sense we should only backport things that are urgent bugs or fixes for regressions that are picked up after the release so that we minimise the chance of introducing any new bugs.

In other words the release branch primarily exists just so that we can assure downstream users that the release matches the release candidate and that any bugfix releases only contain bugfixes.

Reply:
> #24800 is a typical thing that could be backported at this stage

I wouldn't classify that one as having any urgency. That's adding a helpful error message in case someone uses `nsolve` incorrectly which is nice to do but not really what I would call a bugfix. If say it was a new bug since the last release that meant that `nsolve` now returns an incorrect result *when used correctly* then I would say that it should be backported.

Reply:
Yeah, one can discuss what could/should go in... Personally, I'd backport "non-controversial improvements" (for some definition) during the preparation period, but then I am not releasing anything (which is probably good for many other reasons, possibly including this).